causalml/inference/meta/slearner.py

```diff
@@ -219,11 +220,18 @@ def estimate_ate(
             bootstrap_ci (bool): whether to return confidence intervals
             n_bootstraps (int): number of bootstrap iterations
             bootstrap_size (int): number of samples per bootstrap
+            pretrain (bool): whether a model has been fit, default False.
         Returns:
             The mean and confidence interval (LB, UB) of the ATE estimate.
         """
+
         X, treatment, y = convert_pd_to_np(X, treatment, y)
-        te, yhat_cs, yhat_ts = self.fit_predict(X, treatment, y, return_components=True)
+        if pretrain:
+            te, yhat_cs, yhat_ts = self.predict(X, treatment, y, return_components=True)
```

Comment:
QQ: for S-learner if we already have the model, we could has have `y=None` here right? As what you did for R-leaner.

Reply:
Yes you are right, but since predict function here also accepts y, treatment as a parameter, I included them here.